tests/integration/v2/test_integration_v2.py

```diff
@@ -16,112 +20,146 @@
 TLS_CERTIFICATES_REQUIRER_APP_NAME = "tls-certificates-requirer"
 
 
-def copy_lib_content() -> None:
-    shutil.copyfile(src=LIB_DIR, dst=f"{REQUIRER_CHARM_DIR}/{LIB_DIR}")
-    shutil.copyfile(src=LIB_DIR, dst=f"{PROVIDER_CHARM_DIR}/{LIB_DIR}")
-
-
-class TestIntegration:
-    requirer_charm = None
-    provider_charm = None
-
-    async def test_given_charms_packed_when_deploy_charm_then_status_is_blocked(
-        self, ops_test: OpsTest
-    ):
-        assert ops_test.model
-        copy_lib_content()
-        TestIntegration.requirer_charm = await ops_test.build_charm(f"{REQUIRER_CHARM_DIR}/")
-        TestIntegration.provider_charm = await ops_test.build_charm(f"{PROVIDER_CHARM_DIR}/")
-        await ops_test.model.deploy(
-            TestIntegration.requirer_charm,
-            application_name=TLS_CERTIFICATES_REQUIRER_APP_NAME,
-            series="jammy",
+def build_charm(path: Path) -> Path:
+    shutil.copyfile(src=LIB_DIR, dst=path / Path(LIB_DIR))
+    _ = subprocess.run(
+        ["charmcraft", "pack", "--verbose"],
+        check=True,
+        capture_output=True,
+        encoding="utf-8",
+        cwd=path,
+    )
+    return next(path.glob("*.charm"))
+
+
+@pytest.fixture(scope="module")
+def requirer_charm():
+    charm_path = ""
+    charm_path = build_charm(Path(f"{REQUIRER_CHARM_DIR}/").absolute())
+    yield charm_path
+
+
+@pytest.fixture(scope="module")
+def provider_charm():
+    charm_path = ""
+    charm_path = build_charm(Path(f"{PROVIDER_CHARM_DIR}/").absolute())
+    yield charm_path
+
+
+@pytest.fixture(scope="module")
+def juju():
+    with jubilant.temp_model() as juju:
+        juju.wait_timeout = 1000
+        yield juju
+
+
+def test_given_charms_packed_when_deploy_charm_then_status_is_blocked(
+    juju: jubilant.Juju,
+    requirer_charm: Path,
+    provider_charm: Path,
+):
+    juju.deploy(requirer_charm, app=TLS_CERTIFICATES_REQUIRER_APP_NAME)
+    juju.deploy(provider_charm, app=TLS_CERTIFICATES_PROVIDER_APP_NAME)
+    status = juju.wait(
+        lambda status: (
+            status.apps[TLS_CERTIFICATES_REQUIRER_APP_NAME]
+            .units[f"{TLS_CERTIFICATES_REQUIRER_APP_NAME}/0"]
+            .is_blocked
+            and status.apps[TLS_CERTIFICATES_PROVIDER_APP_NAME]
+            .units[f"{TLS_CERTIFICATES_PROVIDER_APP_NAME}/0"]
+            .is_blocked
+        ),
```

Comment:
If both app and unit status are blocked (which is typical), you can use the `jubilant.all_blocked` helper. Or maybe that doesn't work for Juju 2.9?

```python
lambda status: jubilant.all_blocked(status, [TLS_CERTIFICATES_REQUIRER_APP_NAME, TLS_CERTIFICATES_PROVIDER_APP_NAME])
```

Reply:
For some reason, on Juju 2.9, the apps stay in "agent" initializing and do not go to block, so I need this here.